### ISSUE TITLE
UseDataClass false positive on ViewModel

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#5339]
## Steps to Reproduce
```kotlin
class ErrorViewModel(throwable: Throwable) {
	private val location: Array<StackTraceElement> = throwable.stackTrace

	val depth: Int get() = location.count()
	val stack: String
		get() = location.joinToString(separator = "\n") { it.toString().reversed() }
}
```

## Observed Behavior
Flagged.
> Detekt - UseDataClass: The class ErrorViewModel defines no functionality and only holds data. Consider converting it to a data class.

## Expected Behavior
Not flagged - the class has no data, and only defines functionality (inside properties).

## Context
I'm using a WPF-like ViewModel design pattern that exposes public properties.

## Your Environment

* Version of detekt used: 1.21.0

**Comment by schalkms** [comment]
How should this rule detect that this class actually has functionality, in your opinion?

Anyways, could you use something like this?

```Kotlin
data class ErrorViewModel(
    val throwable: Throwable,
    val location: Array<StackTraceElement> = throwable.stackTrace,
    val depth: Int = location.count(),
    val stack: String = location.joinToString(separator = "\n") { it.toString().reversed() }) {
}
```

**Comment by BraisGabin** [comment]
> How should this rule detect that this class actually has functionality, in your opinion?

☝️ this is a really good question. Should we check if a property uses `get()` or `set()`?

**Comment by schalkms** [comment]
I think this is too simple. What if it just gets or sets the property or calls a simple count() function?

> Should we check if a property uses get() or set()?

**Comment by TWiStErRob** [comment]
Ran into this again with a Selenium lazy page object:
```
class ScheduleMovieItem(
	private val root: WebElement
) {

	val startTime: WebElement
		get() = root.findElement(By.className("film-start"))

	val endTime: WebElement
		get() = root.findElement(By.className("film-end"))

	val title: WebElement
		get() = root.findElement(By.className("film-title"))

	val runtime: WebElement
		get() = root.findElement(By.className("film-runtime"))
```

> Should we check if a property uses get() or set()?

Sounds good to me. `get()`/`set()` are sign of logic, they're literally compiled to functions. And functions hold code logic, not data.

> What if it just gets or sets the property or calls a simple count() function?

A simple `count` call is still logic, not trivial:
```
public fun <T> Iterable<T>.count(): Int {
    if (this is Collection) return size
    var count = 0
    for (element in this) checkCountOverflow(++count)
    return count
}
```

Trivial `get() = field` and `set(value) { field = value }` could be ignored, but that would be flagged by other inspections I guess anyway.
![image](https://github.com/detekt/detekt/assets/2906988/ddbcab99-bd24-4329-b8bc-03d9ee9441d9)

**Comment by atulgpt** [comment]
Hi, @9tilov your although related to `UseDataClass` rule, but happening due to some other reason. Can you post the full code and create a new issue(if really not related to this issue)

**Comment by lexa-diky** [comment]
I think this problem could be solved with something like `ClassNameSuppressor` which suppresses rule based on passed class name regexp. This issue is usually occurs within standardized design pattern classes such as view models or dependency holders, they usually have some special naming scheme.
